code/modules/mob/new_player/menu.dm

```diff
@@ -85,23 +85,42 @@
 
 /obj/screen/new_player/selection/join_game
 	name = "Join Game"
-	icon_state = "joingame"
+	icon_state = "unready"
 	screen_loc = "NORTH, CENTER-7"
 
-/obj/screen/new_player/selection/join_game/Click()//no ready system
+/obj/screen/new_player/selection/join_game/Initialize()
 	var/mob/new_player/player = hud.mymob
-	sound_to(player, 'sound/effects/menu_click.ogg')
+	update_icon(player)
+	return ..()
 
-	if(GAME_STATE != RUNLEVEL_GAME)
-		to_chat(player, SPAN_WARNING("The round has either not started yet or already ended."))
-		return
+/obj/screen/new_player/selection/join_game/Click()
+	var/mob/new_player/player = hud.mymob
+	sound_to(player, 'sound/effects/menu_click.ogg')
 
-	if(!player.client.holder)
+	if(!player.client.holder && GAME_STATE > RUNLEVEL_LOBBY)
 		var/dsdiff = config.respawn_menu_delay MINUTES - (world.time - player.respawned_time)
 		if(dsdiff > 0)
 			to_chat(player, SPAN_WARNING("You must wait [time2text(dsdiff, "mm:ss")] before rejoining."))
 			return
-	player.LateChoices() //show the latejoin job selection menu
+
+	if(GAME_STATE <= RUNLEVEL_LOBBY)
+		player.ready = !player.ready
+		to_chat(player, "<span class='notice'>You are now [player.ready ? "ready" : "not ready"].</span>")
+
+	else
+		player.LateChoices() //show the latejoin job selection menu
+
+	update_icon(player)
+
+/obj/screen/new_player/selection/join_game/update_icon(mob/new_player/player)
```

Comment:
this doesn't actually work, I tested locally

Reply:
as in you have to relog for it to take effect

Reply:
Works fine on my end, minus it not automatically updating when round starts